ddl/ddl_api.go

```diff
@@ -2815,7 +2823,11 @@ func (d *ddl) getModifiableColumnJob(ctx sessionctx.Context, ident ast.Ident, or
 		return nil, errors.Trace(err)
 	}
 
-	if err = modifiable(&col.FieldType, &newCol.FieldType); err != nil {
+	if err = modifiable(&col.FieldType, &newCol.FieldType, isColumnWithIndex(col.Name.L, t.Meta().Indices)); err != nil {
+		if strings.Contains(err.Error(), "Unsupported modifying collation") {
```

Comment:
How about use `		if terror.ErrorEqual(err, errUnsupportedModifyCollation) {`

Reply:
I used this code before, but found that `errUnsupportedModify` is equal to `errUnsupportedModifyCharset` because they share the same error code and class.

Any more suggestion?

Reply:
OK, got